core/domain/user_jobs_one_off.py

```diff
@@ -626,3 +628,70 @@ def map(item):
     def reduce(key, values):
         values.sort()
         yield (key, values)
+
+
+class ProfilePictureAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that verifies various aspects of profile_picture_data_url in the
+    UserSettingsModel.
+    """
+
+    @classmethod
+    def enqueue(cls, job_id, additional_job_params=None):
+        # We can raise the number of shards for this job, since it goes only
+        # over one type of entity class.
+        super(ProfilePictureAuditOneOffJob, cls).enqueue(job_id, shard_count=32)
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [user_models.UserSettingsModel]
+
+    @staticmethod
+    def map(model):  # pylint: disable=too-many-return-statements
+        if model.deleted:
+            yield ('DELETED', model.id)
+            return
+
+        if model.username is None:
+            yield ('NOT REGISTERED', model.id)
+            return
+
+        if model.profile_picture_data_url is None:
+            yield ('MISSING PROFILE PICTURE', model.id)
+            return
+
+        try:
+            profile_picture_binary = utils.convert_png_data_url_to_binary(
+                model.profile_picture_data_url)
+        except Exception:
+            yield ('WRONG PROFILE PICTURE DATA URL', model.id)
+            return
+
+        if imghdr.what(None, h=profile_picture_binary) != 'png':
+            yield ('PROFILE PICTURE NOT PNG', model.id)
+            return
+
+        try:
+            # Load the image to retrieve dimensions for later verification.
+            height, width = image_services.get_image_dimensions(
+                profile_picture_binary)
+        except Exception:
+            yield ('CANNOT LOAD PROFILE PICTURE', model.id)
+            return
+
+        if (
+                height != user_services.GRAVATAR_SIZE_PX or
+                width != user_services.GRAVATAR_SIZE_PX
+        ):
+            yield (
+                'PROFILE PICTURE NON STANDARD DIMENSIONS - %s,%s' % (
+                    height, width
+                ),
+                model.id
+            )
+            return
+
+        yield ('CORRECT PROFILE PICTURE', model.id)
```

Comment:
This is a reason to yield "SUCCESS --" -- in reduce(), for "SUCCESS" cases, you just want the counts and not the full list of IDs, to avoid output truncation.

Reply:
Done.